src/signals/incident/components/IncidentForm/index.js

```diff
@@ -17,30 +18,71 @@ class IncidentForm extends React.Component {
   constructor(props) {
     super(props);
 
+    this.state = {
+      loading: false,
+      submitting: false,
+      formAction: '',
+      next: null
+    };
+
     this.setForm = this.setForm.bind(this);
     this.setValues = this.setValues.bind(this);
     this.handleSubmit = this.handleSubmit.bind(this);
+    this.setIncident = this.setIncident.bind(this);
+  }
+
+  static getDerivedStateFromProps(props, prevState) {
+    if (!props.postponeSubmitWhenLoading) {
+      return null;
+    }
+
+    const loading = get(props, props.postponeSubmitWhenLoading);
+    if (loading !== prevState.loading) {
+      return {
+        loading,
+        submitting: !loading ? false : prevState.submitting
+      };
+    }
+
+    return null;
   }
 
-  componentWillReceiveProps(props) {
-    this.setValues(props.incidentContainer.incident);
-    this.form.meta.incident = props.incidentContainer.incident;
+  componentDidUpdate(prevProps, prevState) {
+    this.setValues(this.props.incidentContainer.incident);
+    this.form.meta.incident = this.props.incidentContainer.incident;
+    this.form.meta.submitting = this.submitting;
+    if (this.state.loading !== prevState.loading && !this.state.loading && this.state.next) {
+      defer(() => {
```

Comment:
Why is it necessary to use `defer` here?

Reply:
ja helaas wel. ik heb alles geprobeerd om het zonder te doen. ik sta open voor een oplossing zonder defer want het is een antipatten.

Reply:
het komt door de deler op regel 89